tests/conftest.py

```diff
@@ -70,36 +27,36 @@ def ca_ssl_context(cert_authority: trustme.CA) -> ssl.SSLContext:
     return ctx
 
 
-@pytest.fixture(scope="session")
-def example_org_cert(cert_authority: trustme.CA) -> trustme.LeafCert:
-    return cert_authority.issue_cert("example.org")
+def wait_until_pproxy_serve_on_port(host: str, port: int):
+    while True:
+        try:
+            sock = socket.create_connection((host, port))
+        except ConnectionRefusedError:
+            time.sleep(0.25)
+        else:
+            sock.close()
+            break
 
 
 @pytest.fixture(scope="session")
-def example_org_cert_path(example_org_cert: trustme.LeafCert) -> typing.Iterator[str]:
-    with example_org_cert.private_key_and_cert_chain_pem.tempfile() as tmp:
-        yield tmp
+def proxy_server() -> typing.Iterator[URL]:
+    http_proxy_host = "127.0.0.1"
+    http_proxy_port = 8080
 
+    proc = None
+    try:
+        command_str = f"pproxy -l http://{http_proxy_host}:{http_proxy_port}/"
+        command = shlex.split(command_str)
+        proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
-@pytest.fixture()
-def proxy_server(example_org_cert_path: str) -> typing.Iterator[URL]:
-    """Starts a proxy server on a different thread and yields its origin tuple.
+        wait_until_pproxy_serve_on_port(http_proxy_host, http_proxy_port)
 
-    The server is configured to use a trustme CA and key, this will allow our
-    test client to make HTTPS requests when using the ca_ssl_context fixture
-    above.
+        print(f"HTTP proxy started on http://{http_proxy_host}:{http_proxy_port}/")
```

Comment:
Should we instead let the stdout/stderr of the pproxy process pass through?

Not sure which flag we'd need to pass to `Popen(stdout=..., stderr=...)`, but that way we wouldn't need this print statement, and we'd be able to see the full output using controls built into pytest (such as the `-s` flag).

Reply:
You are right, I was not consistent. Removed the `print` statement and the pipes